lib/glific/flows.ex

```diff
@@ -565,21 +565,26 @@ defmodule Glific.Flows do
     value =
       Flow
       |> where([f], f.organization_id == ^organization_id)
-      |> select([:keywords, :id])
+      |> join(:inner, [f], fr in FlowRevision, on: f.id == fr.flow_id)
+      |> select([f, fr], %{keywords: f.keywords, id: f.id, status: fr.status})
       |> Repo.all(skip_organization_id: true)
       |> Enum.reduce(
         %{},
         fn flow, acc ->
-          Enum.reduce(flow.keywords, acc, fn keyword, acc ->
-            ## Keyword matching for the flow is case insenstive.
-            ## So let's clean the keywords before generating the map.
-            keyword = Glific.string_clean(keyword)
-            Map.put(acc, keyword, flow.id)
-          end)
+          if flow.status == "published", do:
+            Enum.reduce(flow.keywords, acc, fn keyword, acc ->
+              keyword = Glific.string_clean(keyword)
+              Map.put(acc, "published", %{keyword: keyword, id: flow.id})
+            end)
+          if flow.status == "draft", do:
+            Enum.reduce(flow.keywords, acc, fn keyword, acc ->
+              keyword = Glific.string_clean(keyword)
+              Map.put(acc, "draft", %{keyword: keyword, id: flow.id})
+            end)
         end
       )
 
-    organization = Partners.organization(organization_id)
+      organization = Partners.organization(organization_id)
 
     organization =
```

Comment:
Can we also clean up and change the variable name here, it should be called flow_keywords or values or something and not organization